test/val/val_layout_test.cpp

```diff
@@ -491,5 +492,54 @@ TEST_F(ValidateEntryPoint, NoEntryPointWithLinkageCapGood) {
   EXPECT_EQ(SPV_SUCCESS, ValidateInstructions());
 }
 
+// OpModuleProcessed
+
+// Returns the assembly for a module with OpModuleProcessed in
+// many places.
+std::string GetModuleWithModuleProcessed() {
+  return R"(
+           OpCapability Shader
+           OpModuleProcessed "in capabilities"
+           OpCapability Linkage
+           OpMemoryModel Logical GLSL450
+           OpName %voidt "void"
+           OpModuleProcessed "in debug"
+           OpName %funct "func"
+%voidt   = OpTypeVoid
+           OpModuleProcessed "in types"
+%funct   = OpTypeFunction %voidt
+%foo     = OpFunction %voidt None %funct
+           OpModuleProcessed "in function before label"
+%entry   = OpLabel
+           OpModuleProcessed "in block"
+           OpReturn
+           OpModuleProcessed "after last block before function"
+           OpFunctionEnd
+           OpModuleProcessed "at end"
+  )";
+}
+
+using ValidateModuleProcessed = spvtest::ValidateBase<bool>;
+
+TEST_F(ValidateModuleProcessed, OkAnywhereInSPIRV11) {
+  CompileSuccessfully(GetModuleWithModuleProcessed(), SPV_ENV_UNIVERSAL_1_1);
+  EXPECT_EQ(SPV_SUCCESS, ValidateInstructions(SPV_ENV_UNIVERSAL_1_1));
+  EXPECT_THAT(getDiagnosticString(), Eq(""));
+}
+
+TEST_F(ValidateModuleProcessed, OkAnywhereInSPIRV12) {
+  CompileSuccessfully(GetModuleWithModuleProcessed(), SPV_ENV_UNIVERSAL_1_1);
+  EXPECT_EQ(SPV_SUCCESS, ValidateInstructions(SPV_ENV_UNIVERSAL_1_2));
+  EXPECT_THAT(getDiagnosticString(), Eq(""));
+}
+
+TEST_F(ValidateModuleProcessed, FailsIn10) {
```

Comment:
Does this fail on the first `OpModuleProcessed`, or any of them?

Since `OpModuleProcessed` isn't a valid 1.0 opcode I just want to be sure that its invalid everywhere for 1.0, and not just inbetween the `OpFunction` & `OpLabel`.